Source/SplineFollow/Private/Components/SFSplineComponent.cpp

```diff
@@ -101,75 +101,36 @@ void USFSplineComponent::UpdateSpline()
 
 void USFSplineComponent::Serialize( FArchive & archive )
 {
-    if ( archive.IsSaving() )
-    {
-        const auto create_marker = [ & ]< typename T, typename U >( T legacy_marker, U object ) {
-            return FSFSplineMarker( legacy_marker.Name, legacy_marker.ItIsEnabled, legacy_marker.Infos, object );
-        };
-
-        for ( const auto & action_marker : StaticActionMarkers )
-        {
-            auto new_marker = create_marker( action_marker, NewObject< USFSplineMarkerObject_Action >( this ) );
-            Cast< USFSplineMarkerObject_Action >( new_marker.Object )->ActionClass = action_marker.ActionClass->GetOwnerClass();
-            SplineMarkers.Emplace( new_marker );
-        }
-
-        StaticActionMarkers.Empty();
-
-        for ( const auto & level_actor_marker : LevelActorActionMarkers )
-        {
-            auto new_marker = create_marker( level_actor_marker, NewObject< USFSplineMarkerObject_LevelActor >( this ) );
-            Cast< USFSplineMarkerObject_LevelActor >( new_marker.Object )->LevelActor = level_actor_marker.LevelActor;
-            SplineMarkers.Emplace( new_marker );
-        }
-
-        LevelActorActionMarkers.Empty();
-
-        for ( const auto & data_marker : DataMarkers )
-        {
-            auto new_marker = create_marker( data_marker, NewObject< USFSplineMarkerObject_Data >( this ) );
-            Cast< USFSplineMarkerObject_Data >( new_marker.Object )->Data = data_marker.Data;
-            SplineMarkers.Emplace( new_marker );
-        }
+    Super::Serialize( archive );
 
-        DataMarkers.Empty();
-    }
+    StaticActionMarkers.Empty();
+    LevelActorActionMarkers.Empty();
+    DataMarkers.Empty();
+}
 
-    Super::Serialize( archive );
+void USFSplineComponent::CreateMarkerFromDefault( const FSFSplineMarker & default_marker, FSFSplineMarker & new_marker )
```

Comment:
you can just return the constructed object

Reply:
done